### PR TITLE
feat(across-relayer): filter out DAI in cross domain finalizer

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

There is a bug with DAI on optimism with Across. this PR removes dai from the bot to fix this.

### CODE REVIEW

File: packages/insured-bridge-relayer/src/CrossDomainFinalizer.ts [hunk 1]
```diff
@@ -44,8 +44,11 @@ export class CrossDomainFinalizer {
   async checkForBridgeableL2TokensAndBridge() {
     this.logger.debug({ at: "CrossDomainFinalizer", message: "Checking bridgeable L2 tokens" });
 
-    // Fetch all whitelisted tokens on the particular l2 chainId.
-    const whitelistedL2Tokens = this.l1Client.getWhitelistedL2TokensForChainId(this.l2Client.chainId.toString());
+    // Fetch all whitelisted tokens on the particular l2 chainId. Remove the DAI Optimism address from the whitelist as
+    // we don't want to finalize DAI actions due to this not working over the canonical Optimism bridge.
+    const whitelistedL2Tokens = this.l1Client
+      .getWhitelistedL2TokensForChainId(this.l2Client.chainId.toString())
+      .filter((address) => address !== "0xDA10009cBd5D07dd0CeCc66161FC93D7c9000da1"); // Remove DAI on Optimism address.
```
Comment (nicholaspai) [inline packages/insured-bridge-relayer/src/CrossDomainFinalizer.ts:51]: Can you please just add L2_TOKEN_BLACKLIST to the RelayerConfig and filter more upstream from the L1Client? Would be a lot cleaner and easier to change down the line

File: packages/insured-bridge-relayer/src/CrossDomainFinalizer.ts [hunk 2]
```diff
@@ -114,10 +114,11 @@ export class CrossDomainFinalizer {
   async checkForConfirmedL2ToL1RelaysAndFinalize() {
     // Fetch all whitelisted L2 tokens. Append the ETH address on Optimism to the whitelist to enable finalization of
     // Optimism -> Ethereum bridging actions. This is needed as we send ETH over the Optimism bridge, not WETH.
+    // Also, remove the DAI Optimism address from the whitelist as we don't want to finalize DAI actions due to a bug.
     const whitelistedL2Tokens = [
       ...this.l1Client.getWhitelistedL2TokensForChainId(this.l2Client.chainId.toString()),
-      "0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000",
-    ];
+      "0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000", // Append L2ETH on Optimism address.
+    ].filter((address) => address !== "0xDA10009cBd5D07dd0CeCc66161FC93D7c9000da1"); // Remove DAI on Optimism address.
```
[the remark this answers is not in the document]
Comment (chrismaree) [reply packages/insured-bridge-relayer/src/CrossDomainFinalizer.ts:124]: the address that comes back from the L1 client is cased like this.

File: packages/insured-bridge-relayer/src/CrossDomainFinalizer.ts (hunk 1 above)
Comment (chrismaree) [reply packages/insured-bridge-relayer/src/CrossDomainFinalizer.ts:51]: we could do this...but is it worth it? we are going to be deprecating this bot within a short period of time and I don't really think it matters to have some hot glue holding some of it together until we deprecate this. you ok with this?

Comment (nicholaspai) [reply packages/insured-bridge-relayer/src/CrossDomainFinalizer.ts:51]: yea lgtm

**Review by nicholaspai — APPROVED** [review]

**Review by mrice32 — APPROVED** [review]

**Pull request merged**